agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -105,6 +106,36 @@ namespace beerocks {
 #define FSM_IS_IN_STATE(eState) (m_eFSMState == EState::eState)
 #define FSM_CURR_STATE_STR s_arrStates[int(m_eFSMState)]
 
+//////////////////////////////////////////////////////////////////////////////
+/////////////////////////////// Helper Functions /////////////////////////////
+//////////////////////////////////////////////////////////////////////////////
+
+template <class SlavesContainer, class MAC>
+std::shared_ptr<backhaul_manager::sRadioInfo> findRadioInfo(const SlavesContainer &slave_container,
+                                                            const MAC &mac)
```

Comment:
Me also don't think I'll ever have different types here.
However, this is a free function, not part of the backhaul manager. Potentially it has nothing to do with the exact types that are kept in the backhaul manager. Potentially the backhaul manager may decide to keep the data differently.
I keep saying  `potentially` because the amount of the constrains that are applied on the parameters and return value of this function make it unlikely to ever be reused.
Bottom line: I think I'll use the backhaul types for this free function.

Reply:
> Potentially it has nothing to do [...]

I guess you're too young to have been washed with the Extreme Programming principles of [Do The Simplest Thing That Could Possibly Work](http://wiki.c2.com/?DoTheSimplestThingThatCouldPossiblyWork) :-)